From 6589854a4aaaf813192a608242fb71fbff678a09 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Fri, 7 May 2021 09:10:50 -0700
Subject: [PATCH 1522/1921] octeontx2-af: Increase the NIX LF resources for
 98xx

98xx has NIX1, so increase the total limit by adding NIX1 LFs as well.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: I6b4fea3910b548395e536a2e30c5d7a12d96a0ca
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51336
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
index 573239c70ef9..7406503e584b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
@@ -945,7 +945,7 @@ int rvu_policy_init(struct rvu *rvu)
 		goto error;
 	}
 
-	max = hw->block[BLKADDR_NIX0].lf.max;
+	max = hw->block[BLKADDR_NIX0].lf.max + hw->block[BLKADDR_NIX1].lf.max;
 	rvu->pf_limits.nix = quotas_alloc(rvu->hw->total_pfs, max, max,
 					  0, &rvu->rsrc_lock, &pf_limit_ops);
 	if (!rvu->pf_limits.nix) {
-- 
2.31.1

